paper-server/patches/sources/net/minecraft/world/entity/EntitySelector.java.patch

```diff
@@ -23,31 +23,49 @@
++                    return false;
++                } else if (!entity.level().isClientSide || entity1 instanceof Player player && player.isLocalPlayer()) {
++                    Team team1 = entity1.getTeam();
++                    Team.CollisionRule collisionRule1 = team1 == null ? Team.CollisionRule.ALWAYS : team1.getCollisionRule();
++                    if (collisionRule1 == Team.CollisionRule.NEVER || (pushedEntity instanceof Player && !io.papermc.paper.configuration.GlobalConfiguration.get().collisions.enablePlayerCollisions)) { // Paper - Configurable player collision
                          return false;
-                     } else if (!entity.level().isClientSide || entity1 instanceof Player player && player.isLocalPlayer()) {
-                         Team team1 = entity1.getTeam();
-                         Team.CollisionRule collisionRule1 = team1 == null ? Team.CollisionRule.ALWAYS : team1.getCollisionRule();
+-                    } else if (!entity.level().isClientSide || entity1 instanceof Player player && player.isLocalPlayer()) {
+-                        Team team1 = entity1.getTeam();
+-                        Team.CollisionRule collisionRule1 = team1 == null ? Team.CollisionRule.ALWAYS : team1.getCollisionRule();
 -                        if (collisionRule1 == Team.CollisionRule.NEVER) {
-+                        if (collisionRule1 == Team.CollisionRule.NEVER || (pushedEntity instanceof Player && !io.papermc.paper.configuration.GlobalConfiguration.get().collisions.enablePlayerCollisions)) { // Paper - Configurable player collision
-                             return false;
-                         } else {
-                             boolean flag = team != null && team.isAlliedTo(team1);
+-                            return false;
+-                        } else {
+-                            boolean flag = team != null && team.isAlliedTo(team1);
+-                            return (collisionRule != Team.CollisionRule.PUSH_OWN_TEAM && collisionRule1 != Team.CollisionRule.PUSH_OWN_TEAM || !flag)
+-                                && (collisionRule != Team.CollisionRule.PUSH_OTHER_TEAMS && collisionRule1 != Team.CollisionRule.PUSH_OTHER_TEAMS || flag);
+-                        }
+                     } else {
+-                        return false;
++                        boolean flag = team != null && team.isAlliedTo(team1);
++                        return (collisionRule != Team.CollisionRule.PUSH_OWN_TEAM && collisionRule1 != Team.CollisionRule.PUSH_OWN_TEAM || !flag)
++                            && (collisionRule != Team.CollisionRule.PUSH_OTHER_TEAMS && collisionRule1 != Team.CollisionRule.PUSH_OTHER_TEAMS || flag);
+                     }
++                } else {
++                    return false;
+                 }
+-            ));
++            }
++        ));
++        // Paper end - Climbing should not bypass cramming gamerule
```

Comment:
Comment is not necessary, this method is not touched for the climbing cramming bypass option